engine/src/execution_graph/logic_controllers/BatchJoinProcessing.cpp

```diff
@@ -938,89 +956,108 @@ void JoinPartitionKernel::small_table_scatter_distribution(std::unique_ptr<ral::
 	this->output_cache(small_output_cache_name)->wait_for_count(total_count);	
 }
 
-void JoinPartitionKernel::do_process(std::vector<std::unique_ptr<ral::frame::BlazingTable>> inputs,
+ral::execution::task_result JoinPartitionKernel::do_process(std::vector<std::unique_ptr<ral::frame::BlazingTable>> inputs,
 	std::shared_ptr<ral::cache::CacheMachine> /*output*/,
 	cudaStream_t /*stream*/, const std::map<std::string, std::string>& args) {
+	bool input_consumed = false;
+	try{
+		auto& operation_type = args.at("operation_type");
+		auto & input = inputs[0];
+		if (operation_type == "small_table_scatter") {
+			input_consumed = true;
+			std::string small_output_cache_name = scatter_left_right.first ? "output_a" : "output_b";
+			int small_table_idx = scatter_left_right.first ? LEFT_TABLE_IDX : RIGHT_TABLE_IDX;
+
+			broadcast(std::move(input),
+				this->output_.get_cache(small_output_cache_name).get(),
+				"", //message_id_prefix
+				small_output_cache_name, //cache_id
+				small_table_idx //message_tracker_idx
+			);
+		}	else if (operation_type == "big_table_passthrough") {
```

Comment:
The big table passthrough stuff, i think is no longer done in `do_process`. This looks like there was a bad merge somewhere. I think this happens in JoinPartitionKernel::run

Reply:
I didn't find any reference to big_table_passthrough in the main branch, and this is the only reference to this operation_type, so it seems it is an incomplete feature. Should I remove it?

Reply:
Yes, we no longer need this. Please delete